usersimcrs/simulator/agenda_based/interaction_model.py

```diff
@@ -16,31 +18,40 @@ class InteractionModel:
     STOP_INTENT = Intent("STOP")
 
     def __init__(
-        self, config_file, annotated_conversations: List[Dict]
+        self, config_file: str, annotated_conversations: List[Dict[str, Any]]
```

Comment:
Possibly future todo: Can we use a more readable type hint here, specifically `List[Dialogue]`?
(also applies to `intent_distribution()` below)

Reply:
I suppose we can however it would imply that the JSON file with the conversation can be read by DialogueKit and require refactoring of the method `intent_distribution`.

Reply:
I'm planning on refactoring `intent_distribution` so if you could leave a TODO there explaining what'd need to be done plus create an issue in DialogueKit for reading in JSON file with conversations, that'd be good. (Perhaps `Collection[Dialogue]` would be an even better type to indicate that it's iterable but the order doesn't matter.)

Reply:
DialogueKit already has a feature to parse JSON to Dialogue (see [dialogue_reader](https://github.com/iai-group/DialogueKit/blob/main/dialoguekit/utils/dialogue_reader.py)). 
Follow up issue for `intent_distribution` refactoring: #104